src/Microsoft.ServiceFabric.Services/Communication/Client/IRetryPolicy.cs

```diff
@@ -0,0 +1,37 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License (MIT).See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services.Communication.Client
+{
+    using System;
+
+    /// <summary>
+    /// Specifies the retry policy for retrying requests on exceptions in the communication channel between
+    /// client and service replicas.
+    /// </summary>
+    public interface IRetryPolicy
+    {
+        /// <summary>
+        /// Gets the maximum number of times to retry.
+        /// </summary>
+        /// <value>Maximum number of times to retry a specific exception.</value>
+         int TotalNumberOfRetries { get; }
```

Comment:
Is this specific to each exceptipon? This should be the 'total' rite?

Reply:
No. It is specific to each exception.